modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1114,52 +1114,14 @@ public TableImpl table(IgniteUuid id) throws NodeStoppingException {
      * @param checkConfiguration True when the method checks a configuration before tries to get a table, false otherwise.
      * @return A table or {@code null} if table does not exist.
      */
-    private CompletableFuture<Table> tableAsync(String name, boolean checkConfiguration) {
+    private CompletableFuture<? extends Table> tableAsync(String name, boolean checkConfiguration) {
         if (checkConfiguration && !isTableConfigured(name)) {
```

Comment:
This condition does not need here anymore, because we make the same check in `tableAsyncInternal(IgniteUUID)`.
The flag `checkConfiguration` should move to `tableAsyncInternal(IgniteUUID)` and pass false from the method, we find a table by name, that mean we already check that the table exist.